src/Dapr.Client/ConfigurationSource.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Dapr.Client
+{
+    /// <summary>
+    /// Abstraction around a configuration source.
+    /// </summary>
+    public abstract class ConfigurationSource
+    {
+        /// <summary>
+        /// Read any updates that may have occurred in the source configuration. The source may be reading from an ongoing request
+        /// so it utilizes a callback to propagate information back to any concerned parties.
+        /// </summary>
+        /// <param name="callback"><see cref="Action"/> that allows for any changes to propagate back to whatever is watching.</param>
+        /// <param name="cancellationToken">Token used to cancel the operation.</param>
+        /// <returns></returns>
+        public abstract Task ReadSource(Action<ConfigurationItem> callback, CancellationToken cancellationToken = default);
```

Comment:
This type should probably be replaced with an `IAsyncEnumerable<ConfigurationItem>` - that would be "native"

Reply:
Done.